msgpack/_unpacker.pyx

```diff
@@ -101,6 +102,9 @@ cdef inline init_ctx(unpack_context *ctx,
 
     ctx.user.encoding = encoding
     ctx.user.unicode_errors = unicode_errors
+    Py_INCREF(d)
+    ctx.user.memo = <PyObject*>d
```

Comment:
AFAIK `<object>d` will enable refcounting in Cython. Otherwise `object` and `PyObject*` are identical.

Reply:
However, I didn't manage to follow this suggestion, because the `memo` field is in a C struct, which doesn't support `object` fields.